data/types/base/bolus/extended/extended.go

```diff
@@ -28,15 +28,28 @@ func SubType() string {
 	return "square"
 }
 
-func New() (*Extended, error) {
-	extendedBolus, err := bolus.New(SubType())
-	if err != nil {
-		return nil, err
-	}
+func NewDatum() data.Datum {
+	return New()
+}
+
+func New() *Extended {
+	return &Extended{}
+}
+
+func Init() *Extended {
+	extended := New()
+	extended.Init()
+	return extended
+}
+
+func (e *Extended) Init() {
```

Comment:
Just wondering should these be `private`? We call the `func Init() *Extended {...}` which internally calls `func (e *Extended) Init() {...}` but would we ever call `func (e *Extended) Init() {...}` on its own?